Add Vagrantfile for development environment bootstrapping

ewpatton opened the pull request:
Change-Id: I2fb08206d96f2ff0da73954f820134b7a0684aef

ewpatton commented:
The initial version of this was implemented by @nicholec. I simplified it a bit and updated it to use Ubuntu 18.04 as the base image instead of 16.04.

@nicholec: I marked you as the author. If you'd prefer that not to be the case, let us know.

@SusanRatiLane @jisqyv: When we go to merge this, don't do it through the GitHub interface. The problem is that GitHub changes the author of the commit to the pull request submitter and makes the committer the person who pushed the merge button. Instead, we should push it from the command line to preserve the authorship information.

@halatmit @josmas: I'm ccing you because you might be interested in testing this out.

josmas commented:
Hey @ewpatton a few comments on this.

I saw the following errors during install, but it did not stop the compilation
```
    default: update-alternatives: error: no alternatives for jaotc
    default: update-alternatives: error: no alternatives for jdeprscan
    default: update-alternatives: error: no alternatives for jhsdb
    default: update-alternatives: error: no alternatives for jimage
    default: update-alternatives: error: no alternatives for jlink
    default: update-alternatives: error: no alternatives for jmod
    default: update-alternatives: error: no alternatives for jshell
    default: update-alternatives: error: no alternatives for mozilla-javaplugin.so
    default: update-java-alternatives: plugin alternative does not exist: /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/amd64/IcedTeaPlugin.so
```

The `ant` command took over 17 minutes, which is a bit odd. I guess I can give the VM more RAM or something. In any case, the instructions mention a `start_appinventor` script, but I cannot find that.

This [Dockerfile](https://github.com/josmas/appinventor-env-docker) is quite old now and I haven't updated it in 3 years, but there are a few things in there that might be worth considering here, such as installing _adb_ and adding rules for USB in order to plug devices, and also installing _phantomJS_ to run tests.

jisqyv commented on Vagrantfile:
```diff
@@ -0,0 +1,29 @@
+ 
+  config.vm.provider "virtualbox" do |v|
+    v.default_nic_type = "Am79C973"
+    v.name = "ForAppinventor2-xenial64"
```
Should call this ForAppinventor2-bionic64

jisqyv commented on Vagrantfile:
```diff
@@ -0,0 +1,29 @@
+  config.vm.boot_timeout = 400 
+ 
+  config.vm.provider "virtualbox" do |v|
+    v.default_nic_type = "Am79C973"
```
This line results in errors under Linux. Commenting it out solves the problem

ewpatton commented:
I added adb and phantomjs to the apt-get command, and removed the networking lines causing issues on Linux. I experimented with some different approaches to improving the disk access speed, but based on what I've read this is a problem generally with the VirtualBox shared folders mechanism, which Vagrant relies on. Some people recommend switching to NFS, but that isn't really viable for Windows users. Others say that VMWare and Parallels provide better performance. I'm willing to say that performance shouldn't be a primary concern because people can always follow the manual instructions to set up their dev workstations appropriately.

SusanRatiLane commented:
Works for me on Mojave with VirtualBox.
